snapd.go

```diff
@@ -479,14 +505,27 @@ func action(ctx *cli.Context) {
 		r.BindMetricManager(c)
 		r.BindConfigManager(c.Config)
 		r.BindTaskManager(s)
+		//Rest Authentication
+		if restAuth {
+			log.Info("REST API authentication is enabled")
+			r.SetAPIAuth(restAuth)
+			// restAuthPwd provided in config file
+			password := []byte(restAuthPwd)
+			log.Info("REST API authentication password is set")
+			hashedPwd, err := bcrypt.GenerateFromPassword(password, bcrypt.DefaultCost)
+			if err != nil {
+				log.Fatal("Unable to hash password from snapctl: ", err.Error)
```

Comment:
This error message needs to be changed. snapctl plays no part in snapd starting the different modules of the system.